apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/CarrierClaimTransformerV2Test.java

```diff
@@ -1228,6 +1230,89 @@ public void shouldHaveProcedureList() {
     assertEquals(0, eob.getProcedure().size());
   }
 
+  /** Test that should not have a npi entry for organization. */
+  @Test
+  public void shouldNotHaveNpiEntryForOrg() throws IOException {
+    List<Object> parsedRecords =
+        ServerTestUtils.parseData(Arrays.asList(StaticRifResourceGroup.SAMPLE_A.getResources()));
+
+    CarrierClaim claimWithoutNpi =
+        parsedRecords.stream()
+            .filter(r -> r instanceof CarrierClaim)
+            .map(r -> (CarrierClaim) r)
+            .findFirst()
+            .get();
+
+    claimWithoutNpi.setLastUpdated(Instant.now());
+    claimWithoutNpi.getLines().get(0).setOrganizationNpi(Optional.empty());
+    ExplanationOfBenefit genEob =
+        CarrierClaimTransformerV2.transform(
+            new TransformerContext(
+                new MetricRegistry(),
+                Optional.empty(),
+                FdaDrugCodeDisplayLookup.createDrugCodeLookupForTesting(),
+                NPIOrgLookup.createNpiOrgLookupForTesting()),
+            claimWithoutNpi);
+    IParser parser = fhirContext.newJsonParser();
+    String json = parser.encodeResourceToString(genEob);
+    ExplanationOfBenefit eobWithoutNpi = parser.parseResource(ExplanationOfBenefit.class, json);
+
+    C4BBPractitionerIdentifierType type = C4BBPractitionerIdentifierType.NPI;
+    C4BBClaimProfessionalAndNonClinicianCareTeamRole role =
+        C4BBClaimProfessionalAndNonClinicianCareTeamRole.PRIMARY;
+
+    CareTeamComponent careTeamEntry =
+        eobWithoutNpi.getCareTeam().stream()
+            .filter(
+                ctc ->
+                    ctc.getProvider().getIdentifier().getType().getCoding().stream()
+                        .anyMatch(
+                            c ->
+                                c.getSystem().equalsIgnoreCase(type.getSystem())
+                                    && c.getCode().equalsIgnoreCase(type.toCode())))
+            .filter(
+                ctc ->
+                    role.toCode().equalsIgnoreCase(ctc.getRole().getCodingFirstRep().getCode())
+                        && role.getSystem()
+                            .equalsIgnoreCase(ctc.getRole().getCodingFirstRep().getSystem()))
+            .findAny()
+            .orElse(null);
+
+    assertEquals(null, careTeamEntry);
+  }
+
+  /** Test that should have a npi entry for organization. */
+  @Test
+  public void shouldHaveANpiEntryForOrg() throws IOException {
```

Comment:
Same here, test that we should have an npi entry for organization _when the claim's organization npi is present_

Reply:
done